File: MovieDB/app/src/main/java/com/tp/projects/moviedb/movies/MovieDetailsActivity.java
package com.tp.projects.moviedb.movies;

import android.content.Context;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;
import com.tp.projects.moviedb.R;

public class MovieDetailsActivity extends AppCompatActivity {

  Context ctx;
  MovieData movie;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_movie_details);
    Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
    setSupportActionBar(toolbar);
    if (getIntent().getExtras() != null) {
      movie = (MovieData) getIntent().getExtras().get("movie");
    }
    toolbar.setTitle(movie.getTitle());
    toolbar.setNavigationIcon(R.drawable.ic_action_navigation_arrow_back);
    toolbar.setNavigationOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        onBackPressed();
      }
    });
    ctx = this;
    ImageView header = (ImageView) findViewById(R.id.movie_detail_header);
    setUI(movie);
    Picasso.with(ctx)
      .load(movie.getBackdropPath())
      .error(R.mipmap.ic_launcher)
      .into(header);
    if (findViewById(R.id.value_adult) != null) {
      ((TextView) findViewById(R.id.value_adult)).setText(movie.getAdult() ? "Yes" : "No");
    }
  }

  public void setUI(MovieData movie) {
    ((TextView) findViewById(R.id.value_title)).setText(movie.getTitle());
    ((TextView) findViewById(R.id.value_date)).setText(movie.getReleaseDate());
    ((TextView) findViewById(R.id.value_orig_title)).setText(movie.getOriginalTitle());
    ((TextView) findViewById(R.id.value_overview)).setText(movie.getOverview());
    ((TextView) findViewById(R.id.value_popularity)).setText(movie.getPopularity().toString());
    ((TextView) findViewById(R.id.value_vote_avg)).setText(movie.getVoteAverage().toString());
  }
}

File: MovieDB/app/src/main/java/com/tp/projects/moviedb/movies/MovieDetailsActivity.kt
package com.tp.projects.moviedb.movies

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.widget.ImageView
import android.widget.TextView
import com.squareup.picasso.Picasso
import com.tp.projects.moviedb.R

class MovieDetailsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_movie_details)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        val header = findViewById<ImageView>(R.id.movie_detail_header)
        setSupportActionBar(toolbar)

        intent.extras?.get("movie")?.let {
            toolbar.title = (it as MovieData).title
            setUI(it)
            Picasso.get().load(it.backdropPath).error(R.mipmap.ic_launcher).into(header)
            findViewById<TextView>(R.id.value_adult).text = if (it.adult) getString(android.R.string.yes) else getString(android.R.string.no)
        }
        toolbar.apply {
            setNavigationIcon(R.drawable.ic_action_navigation_arrow_back)
            setNavigationOnClickListener { onBackPressed() }
        }
    }

    fun setUI(movie: MovieData) {
        movie.apply {
            findViewById<TextView>(R.id.value_title).text = title
            findViewById<TextView>(R.id.value_date).text = releaseDate
            findViewById<TextView>(R.id.value_orig_title).text = originalTitle
            findViewById<TextView>(R.id.value_overview).text = overview
            findViewById<TextView>(R.id.value_popularity).text = popularity.toString()
            findViewById<TextView>(R.id.value_vote_avg).text = voteAverage.toString()

        }
    }
}
